pkg/provider/cloud/openstack/provider.go

```diff
@@ -294,17 +294,21 @@ func reconcileNetwork(ctx context.Context, netClient *gophercloud.ServiceClient,
 		networkName = cluster.Name
 	}
 
-	newNetwork, err := createUserClusterNetwork(netClient, networkName)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create the kubermatic network: %w", err)
-	}
-	cluster, err = update(ctx, cluster.Name, func(cluster *kubermaticv1.Cluster) {
+	cluster, err := update(ctx, cluster.Name, func(cluster *kubermaticv1.Cluster) {
 		kubernetes.AddFinalizer(cluster, NetworkCleanupFinalizer)
-		cluster.Spec.Cloud.Openstack.Network = newNetwork.Name
+		cluster.Spec.Cloud.Openstack.Network = resourceNamePrefix + networkName
```

Comment:
Wouldn't this preprend the prefix again and again to the Network field? In line 277 you get the current Network name from the Cluster object and then in 299 you preprend the prefix and update the cluster object. Won't this turn foo into prefix_foo into prefix_prefix_foo?

Reply:
No! Basically if networkName is empty then foo it will be `prefix-clusterID`. `newNetwork.Name` is the name of the created network by the `createUserClusterNetwork` which returns `prefix-networkName`. Since we're updating the cluster before calling the function that create this network. we need to concat prefix with `networkName` when we update the cluster object

Reply:
Yeah, on the _first_ iteration. But once the cluster object is reconciled again, then line 277 will read the current, full network name ("prefix-blablabla") and store it in `networkName`. The if statement in line 280 is true, but does not change the `networkName` variable. Line 293 is then not true (the name is not empty) and so will not be entered. Then we're at the `update()` call and there you overwrite the network name with "prefix-" + "prefix-blablabla". Did you mean to use `network.Name`?